components/ai_rewriter/resources/page/components/Layout.tsx

```diff
@@ -75,38 +65,41 @@ const Root = styled.div`
 `
 
 export default function Layout() {
-  const { undo, canUndo, redo, canRedo, erase, canErase, generate, close, openSettings } = useRewriterContext()
+  const context = useRewriterContext()
   return <Root>
     <TitleRow>
       <Icon name='product-brave-leo' />
       <TitleText>leo writer</TitleText>
       <Label color='blue'>PREMIUM</Label>
       <DialogControls>
-        <Button fab kind='plain-faint' size='small' onClick={openSettings}>
+        <Button fab kind='plain-faint' size='small' onClick={context.openSettings}>
           <Icon name='settings' />
         </Button>
-        <Button fab kind='plain-faint' size='small' onClick={close}>
+        <Button fab kind='plain-faint' size='small' onClick={context.close}>
           <Icon name='close' />
         </Button>
       </DialogControls>
     </TitleRow>
-    <FiltersContainer>
-      <InitialText />
-    </FiltersContainer>
-    <NoContent />
+    {context.generatedText
+      ? <ModifyGeneration />
+      : <BeginGeneration />}
     <ActionsRow>
-      <Button fab kind="outline" isDisabled={!canUndo} onClick={undo}>
+      <Button fab kind="outline" isDisabled={!context.canUndo} onClick={context.undo}>
         <Icon name="arrow-undo" />
       </Button>
-      <Button fab kind="outline" isDisabled={!canRedo} onClick={redo}>
+      <Button fab kind="outline" isDisabled={!context.canRedo} onClick={context.redo}>
         <Icon name="arrow-redo" />
       </Button>
-      <Button fab kind="outline" isDisabled={!canErase} onClick={erase}>
+      <Button fab kind="outline" isDisabled={!context.canErase} onClick={context.erase}>
         <Icon name="erase" />
       </Button>
       <FlexSpacer />
-      <Button onClick={generate}>
-        Generate
+      <Button isLoading={context.isGenerating} onClick={context.generatedText ? context.acceptGeneratedText : context.submitRewriteRequest} isDisabled={context.isGenerating}>
```

Comment:
I'm wondering if we should also disable the button when
* prompt is empty
* tool isn't selected.